### ISSUE TITLE
Throw customized message

### ISSUE THREAD

**Issue by binchensjtu** [fge/json-schema-validator#140]
It is better to support customized error message after failed.
```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "id": "http://jsonschema.net",
  "type": "object",
  "title": "Root schema.",
  "description": "address is required",
  "properties": {
    "address": {
      "id": "http://jsonschema.net/address",
      "type": "object",
      "title": "Address schema.",
      "description": "address is invalid",
      "properties": {
        "streetAddress": {
          "id": "http://jsonschema.net/address/streetAddress",
          "type": "string",
          "title": "StreetAddress schema.",
          "description": "streetAddress is invalid"
        },
        "city": {
          "id": "http://jsonschema.net/address/city",
          "type": "string",
          "title": "City schema.",
          "description": "city is invalid"
        }
      }
    }
  },
  "required": [
    "address"
  ]
}
```
It will throw the most precisely description while validation failed.

**Comment by fge** [comment]
Sorry, I don't really understand your question?

Also, `description` is not meant for holding error messages; don't forget that by default you can add any field to a JSON Schema; extraneous fields are just ignored. You could for instance define `errorMessage`.

**Comment by aschrijver** [comment]
Yes indeed, I think `description` should not be constrained to just validation messages, that is not its purpose. Its to add descriptive text to you schema which might be used to decorate a user interface, as the specification suggests: http://json-schema.org/latest/json-schema-validation.html#anchor98

**Comment by binchensjtu** [comment]
Sorry I didn't describe my question clearly.

I don't mean it must use the `description`. Maybe define a field `errorMessage` to indicate the error message. It would be some readable message relate to business like 'city must be Washington or New York' instead of 'instance value ("DCdddd") not found in enum (possible values: ["Washington","New York"])'

**Comment by fge** [comment]
Well, the API currently has no support for this. What you _can_ do however is override the error messages for a particular keyword; but that will apply to each and every instance of this keyword in your schema...
